### PR TITLE
[GOBBLIN-748]Craftsmanship code cleaning in Gobblin Service Code

### PR DESCRIPTION
**Pull request by autumnust**
Dear Gobblin maintainers,

Please accept this PR. I understand that it will not be reviewed until I have checked off all the steps below!

Essentially a bunch of renaming, moving configurations key around. 
Things that we should agree on: 
- Configuration keys should always placed in the class where they are relevant, instead of `ConfigurationKeys.java` or similar things. We should definitely stop doing that. 
- Naming of `FSFlowCatalog` is misleading. In fact it should be `FsTemplateFlowCatalog`. Also, it should not extend `FsJobCatalog`. The missing piece is a `FsCatalogBase`  that implement all FileSystem related basic functionality, and have `Template` and `Job` FsCatalog extending that base implementation with it own interface. 

### JIRA
- [x ] My PR addresses the following [Gobblin JIRA]
    - https://issues.apache.org/jira/browse/GOBBLIN-748

### Description
- [ x] Here are some details about my PR, including screenshots (if applicable):

### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### CODE REVIEW

**Comment by autumnust** [comment]
@sv2000 @jack-moseley Can you take a look ? Thanks

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/template_catalog/FSFlowTemplateCatalog.java [hunk 1]
```diff
@@ -125,13 +123,7 @@ public FlowTemplate getFlowTemplate(URI flowTemplateDirURI)
...
     };
 
-    if (!this.sysConfig.hasPath(ServiceConfigKeys.TEMPLATE_CATALOGS_FULLY_QUALIFIED_PATH_KEY)) {
-      throw new RuntimeException("Missing config " + ServiceConfigKeys.TEMPLATE_CATALOGS_FULLY_QUALIFIED_PATH_KEY);
-    }
-    if (!flowTemplateDirURI.getScheme().equals(FS_SCHEME)) {
-      throw new RuntimeException(
-          "Expected scheme " + FS_SCHEME + " got unsupported scheme " + flowTemplateDirURI.getScheme());
-    }
+    validateTemplateURI(flowTemplateDirURI);
```
Comment (sv2000) [inline gobblin-service/src/main/java/org/apache/gobblin/service/modules/template_catalog/FSFlowTemplateCatalog.java]: Throw an exception if flowTemplateDirUri is invalid. Change it to if (!isValidTemplateURI()) {...}.

**Comment by sv2000** [comment]
A few more clean up asks while you are at it:
1. Can you fix the Javadoc references (i.e. missing "@link") in some of the classes? e.g. FlowEdge, BaseFlowGraph. The build shows which of the classes has the references broken. 
2. What is the plan for MultiHopFlowToJobSpecCompiler and the associated classes? If those classes are broken, should we delete them too as part of the clean up?

**Comment by autumnust** [comment]
> A few more clean up asks while you are at it:
> 
> 1. Can you fix the Javadoc references (i.e. missing "@link") in some of the classes? e.g. FlowEdge, BaseFlowGraph. The build shows which of the classes has the references broken.
> 2. What is the plan for MultiHopFlowToJobSpecCompiler and the associated classes? If those classes are broken, should we delete them too as part of the clean up?

Will fix the first point and open another PR to address the second. It seems there are a lot of redundancy existed for `SpecCompiler`.

File: gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java [hunk 2]
```diff
@@ -357,10 +357,9 @@ private void initialize(Dag<JobExecutionPlan> dag)
     }
 
     /**
-     * Poll the statuses of running jobs.
-     * @return List of {@link JobStatus}es.
+     * Proceed the execution of each dag node based on job status.
      */
-    private void pollJobStatuses()
+    private void proceedDagExecutionOnDagNodeStatus()
```
[the remark this answers is not in the document]
Comment (autumnust) [reply gobblin-service/src/main/java/org/apache/gobblin/service/modules/orchestration/DagManager.java]: +1

**Comment by autumnust** [comment]
Created Ticket: [GOBBLIN-754 ](https://issues.apache.org/jira/browse/GOBBLIN-754) to track cleaning mentioned in comments above.

**Comment by sv2000** [comment]
@autumnust there seem to be conflicting changes. Can you merge the conflicts?

**Pull request closed without merging**